pkg/logging/logging.go

```diff
@@ -61,6 +61,23 @@ func (l *Logger) NoFields() *logrus.Entry {
 }
 
 func (l *Logger) WithFields(fields logrus.Fields) *logrus.Entry {
+	// If an err field is provided, and it conforms to the util.StackError
+	// interface, add some fields to the log message that can be used for
+	// exception reporting by a buddy launchable (e.g. to bugsnag or
+	// airbrake)
+	var stackErr util.StackError
+	if err, ok := fields["err"]; ok {
```

Comment:
maybe it would be better to test all fields for conformity to the `util.StackError` interface? that way `err` isn't a magic key